diffusion/datasets/image_caption.py

```diff
@@ -73,38 +72,54 @@ def __init__(
         if caption_selection not in ['first', 'random']:
             raise ValueError(f'Invalid caption selection: {caption_selection}. Must be one of [random, first]')
 
+        self.crop = crop
         self.transform = transform
         self.sdxl = sdxl
-        if self.sdxl:
-            self.tokenizer = SDXLTokenizer(tokenizer_name_or_path)
-            self.sdxl_crop = RandomCropSquareReturnTransform(image_size)
-        else:
-            self.tokenizer = AutoTokenizer.from_pretrained(tokenizer_name_or_path, subfolder='tokenizer')
-            self.sdxl_crop = None
         self.caption_drop_prob = caption_drop_prob
         self.microcond_drop_prob = microcond_drop_prob
         self.caption_selection = caption_selection
-        self.image_size = image_size
         self.image_key = image_key
         self.caption_key = caption_key
 
+        if self.sdxl:
+            self.tokenizer = SDXLTokenizer(tokenizer_name_or_path)
+        else:
+            self.tokenizer = AutoTokenizer.from_pretrained(tokenizer_name_or_path, subfolder='tokenizer')
+
     def __getitem__(self, index):
         sample = super().__getitem__(index)
+        out = {}
 
         # Image
         img = sample[self.image_key]
         if not isinstance(img, Image.Image):
             img = Image.open(BytesIO(sample[self.image_key]))
         if img.mode != 'RGB':
             img = img.convert('RGB')
+        orig_h, orig_w = img.size
```

Comment:
According to PIL Image [docs](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.size), `img.size` will return (width, height) tuple, so `orig_h` and `orig_w` should be flipped.